tests/returntypes/paramshift/test_parameter_shift_new.py

```diff
@@ -3292,3 +3292,1034 @@ def test_jax(self, tol, broadcast):
+            qml.RY(y, wires=[1])
+            qml.CNOT(wires=[0, 1])
+            return qml.expval(qml.PauliZ(0)), qml.expval(qml.PauliX(1))
+
+        x = jax.numpy.array([0.543, 0.2])
+        y = jax.numpy.array(-0.654)
+
+        res = qml.gradients.param_shift(circuit, argnums=[0, 1])(x, y)
+
+        res_expected = jax.jacobian(circuit, argnums=[0, 1])(x, y)
+
+        assert np.allclose(res[0][0], res_expected[0][0], atol=tol)
+        assert np.allclose(res[0][1], res_expected[0][1], atol=tol)
+        assert np.allclose(res[1][0], res_expected[1][0], atol=tol)
+        assert np.allclose(res[1][1], res_expected[1][1], atol=tol)
+
+    @pytest.mark.parametrize("interface", interfaces)
+    def test_multi_meas_shape_values_multi_measurements(self, interface, tol):
+        """Test for multiple measurements (with shape) and multiple params."""
+        import jax
+
+        dev = qml.device("default.qubit", wires=2)
+
+        @qml.qnode(dev, interface=interface)
+        def circuit(x, y):
+            qml.RX(x[0], wires=[0])
+            qml.RY(y, wires=[1])
+            qml.CNOT(wires=[0, 1])
+            return qml.expval(qml.PauliZ(0)), qml.probs(wires=[0, 1])
+
+        x = jax.numpy.array([0.543, 0.2])
+        y = jax.numpy.array(-0.654)
+
+        res = qml.gradients.param_shift(circuit, argnums=[0, 1])(x, y)
+
+        res_expected = jax.jacobian(circuit, argnums=[0, 1])(x, y)
+
+        assert np.allclose(res[0][0], res_expected[0][0], atol=tol)
+        assert np.allclose(res[0][1], res_expected[0][1], atol=tol)
+        assert np.allclose(res[1][0], res_expected[1][0], atol=tol)
+        assert np.allclose(res[1][1], res_expected[1][1], atol=tol)
+
+    @pytest.mark.parametrize("interface", interfaces)
+    def test_identity_classical_jacobian(self, interface, tol):
+        """Test for multiple measurements (with shape) and multiple params."""
+        import jax
+
+        dev = qml.device("default.qubit", wires=2)
+
+        @qml.qnode(dev, interface=interface)
+        def circuit(x):
+            qml.RX(x[0], wires=[0])
+            qml.RY(x[1], wires=[1])
+            qml.CNOT(wires=[0, 1])
+            return qml.expval(qml.PauliZ(0))
+
+        x = jax.numpy.array([0.543, -0.654])
+        res = qml.gradients.param_shift(circuit)(x)
+        res_expected = jax.jacobian(circuit)(x)
+
+        assert np.allclose(res[0], res_expected[0], atol=tol)
+        assert np.allclose(res[1], res_expected[1], atol=tol)
+
+
+@pytest.mark.jax
+class TestQnodeJaxJit:
+    """Class to the parameter shift transform with some classical processing."""
+
+    expected_jacs = []
+    interfaces = ["auto", "jax-jit"]
+
+    @pytest.mark.parametrize("interface", interfaces)
+    def test_single_measurement_single_param(self, interface):
+        """Test for a single measurement and a single param."""
+        dev = qml.device("default.qubit", wires=2)
+
+        import jax
+
+        @qml.qnode(dev, interface=interface, cache=False)
```

Comment:
So passing `"jax-jit"` as the interface does the same as if we'd done `@jax.jit`?

Reply:
Out of curiosity, what happens if you use the decorator on the transform, but you set the interface to "jax" instead of "jax-jit"?

Reply:
Nothing for the QNode transform setting it to jax-jit does not change anything. For the the QNode itself it will change nothing as well because there is an internal function to switch to "jax-jit". So it is not necessary at all I think